packages/react-components/react-table/src/hooks/useKeyboardResizing.ts

```diff
@@ -64,25 +64,27 @@ export function useKeyboardResizing(columnResizeState: ColumnResizeState) {
     (colId: TableColumnId) => {
       setColumnId(colId);
       onChangeRef.current?.(colId, true);
-      // This needs to happen asynchronously, because we cannot focus an element with without tabIndex.
-      // Only after we have called "setColumnId" the tabIndex will be set to 0 for this handle.
-      // Meaning we have to wait until the DOM updates with the new tabIndex and then we can focus.
-      setTimeout(() => {
-        resizeHandleRefs.get(colId)?.current?.focus();
-      }, 50);
+
+      const handle = resizeHandleRefs.get(colId)?.current;
+      if (handle) {
+        handle.tabIndex = 0;
+        handle.focus();
+      }
     },
     [resizeHandleRefs],
   );
 
   const disableInteractiveMode = React.useCallback(() => {
-    // Notify the onChange listener that we are disabling interactive mode.
-    if (columnId) {
-      onChangeRef.current?.(columnId, false);
+    if (!columnId) {
+      return;
     }
+    // Notify the onChange listener that we are disabling interactive mode.
+    onChangeRef.current?.(columnId, false);
     // Find the previous focusable element (table header button) and focus it.
     const el = resizeHandleRefs.get(columnId)?.current;
     if (el) {
-      findPrevFocusable(el)?.focus();
+      el.tabIndex = -1; // Make sure the resize handle is not focusable anymore.
```

Comment:
I was doing a bit of playing around and found out that using `element.tabIndex` is quite odd because for an element like a div you can't really 'unset' it. A consequence of that is that once you run `el.tabIndex = -1` for a div, it will **always focus after clicking** and there's no easy way to revert that behaviour https://codesandbox.io/s/nervous-leaf-wyk2j2?file=/src/index.js

Let's do instead `element.setAttribute("tabindex", "-1")` and remove the attribute. It seems safer and after removing the attribute you can no longer focus on that element by clicking on it.

Reply:
Also setting tabIndex = 0 isn't smth we really want since that makes the element focusable using `Tab` we just want the `-1` behaviour here so we can programmatically focus on it using `.focus`

Reply:
Just to be clear we should do

- `element.setAttribute("tabindex", "-1")` to enable keyboard resizing
- `element.removeAttribute("tabindex") to disable keyboard resizing